handler/producer/request.go

```diff
@@ -40,22 +41,32 @@ func (r Requests) Produce(ctx context.Context, _ *http.Request, results chan<- *
 		}
 	}()
 
+	evalctx := ctx.Value(eval.ContextType).(*eval.Context)
+	updated := evalctx.WithClientRequest(req)
+
 	for _, or := range r {
 		outCtx := withRoundTripName(ctx, or.Name)
 
-		method, err := eval.GetContextAttribute(or.Context, outCtx, "method")
+		content, _, diags := or.Context.PartialContent(config.Request{Remain: or.Context}.Schema(true))
+		if diags.HasErrors() {
+			results <- &Result{Err: diags}
+			wg.Done()
```

Comment:
just use `sendResult` here which will handle context cancels too.
`wg.Done()` would result in a negative waitgroup counter which will panic since the `roundtrip` below will handle the done call. Also `wg.Add` gets called next to roundtrip as late as possible due to possible eval error cases.